blockonomics.php

```diff
@@ -200,9 +202,27 @@ public function hookPaymentOptions($params)
     public function getPaymentOption()
     {
         $offlineOption = new PaymentOption();
+        $active_cryptos = $this->getActiveCurrencies();
+        $logoHeight = Configuration::get('BLOCKONOMICS_LOGO_HEIGHT');
+
+        $this->context->smarty->assign('blockonomicsLogoHeight', $logoHeight);
+        
+        $cryptos = array();
+        $logo_icons = array();
+        foreach ($active_cryptos as $crypto) {
+            array_push($cryptos, $crypto['name']);
+            array_push($logo_icons, 
+                _MODULE_DIR_.'blockonomics/views/img/'.$crypto['code'].'-icon.png'
+            );
+        }
+
+        $this->context->smarty->assign('blockonomicsEnabledLogos', $logo_icons);
+        
         $offlineOption
-            ->setCallToActionText($this->l('Pay by bitcoin'))
-            ->setLogo(_MODULE_DIR_.'blockonomics/views/img/bitcoin-icon.png')
+            ->setCallToActionText($this->l('Pay by ' . join(' or ', $cryptos)))
```

Comment:
We can add `->setModuleName('blockonomics')` here, which in turn adds `data-module-name="blockonomics"` to the input checkbox on the checkout. This will then offer a more reliable way to target the correct container. eg
```
let payment_number = document.querySelector("input[data-module-name='blockonomics']").id.split("payment-option-")[1];
let payment_container = document.getElementById("payment-option-" + payment_number + "-container");
{foreach $blockonomicsEnabledLogos as $logo}
    img = document.createElement("img")
    img.src = "{$logo}"
    img.style.height = "{$blockonomicsLogoHeight|intval}px"
    img.style.paddingRight = "5px"
    payment_container.querySelector('label').appendChild(img)
{/foreach}
```

Reply:
Okay. Let me try.

Reply:
The query will then become:

```
let container = document.querySelector("input[data-module-name=blockonomics]").parentElement.parentElement.querySelector("label")

{foreach $blockonomicsEnabledLogos as $logo}
    img = document.createElement("img")
    img.src = "{$logo}"
    img.style.height = "{$blockonomicsLogoHeight|intval}px"
    img.style.paddingRight = "5px"
    container.appendChild(img)
{/foreach}
```